azure-kusto-data/azure/kusto/data/client.py

```diff
@@ -558,6 +563,43 @@ def to_json(self):
         return json.dumps({"Options": self._options, "Parameters": self._parameters}, default=str)
 
 
+class ExecuteRequestParams:
+    def __init__(self, database: str, payload: io.IOBase, properties: ClientRequestProperties, query: str, timeout: timedelta, request_headers: dict):
+        request_headers = copy(request_headers)
+        json_payload = None
+        if not payload:
+            json_payload = {"db": database, "csl": query}
+            if properties:
+                json_payload["properties"] = properties.to_json()
+
+            client_request_id_prefix = "KPC.execute;"
+            request_headers["Content-Type"] = "application/json; charset=utf-8"
+        else:
+            if properties:
+                request_headers.update(json.loads(properties.to_json())["Options"])
+
+            client_request_id_prefix = "KPC.execute_streaming_ingest;"
+            request_headers["Content-Encoding"] = "gzip"
+        request_headers["x-ms-client-request-id"] = client_request_id_prefix + str(uuid.uuid4())
+        if properties is not None:
+            if properties.client_request_id is not None:
+                request_headers["x-ms-client-request-id"] = properties.client_request_id
+            if properties.application is not None:
+                request_headers["x-ms-app"] = properties.application
+            if properties.user is not None:
+                request_headers["x-ms-user"] = properties.user
+            if properties.get_option(ClientRequestProperties.no_request_timeout_option_name, False):

```

Comment:
https://github.com/Azure/azure-kusto-python/blob/a6e41842810ab2b82e81a826dd91fd5f9a6e70a2/azure-kusto-data/azure/kusto/data/client.py#L535

Reply:
This logic was here before the PR, but I assume that this option can be set, but set to False, so has_option isn't enough